modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/shibari_stand.dm

```diff
@@ -0,0 +1,269 @@
+	var/mob/living/buckledmeme = buckled_mob
+	if(buckledmeme)
+		if(buckledmeme != user)
+			if((HAS_TRAIT(user, TRAIT_RIGGER)))
+				if(!do_after(user, 5 SECONDS, buckledmeme))
+					return FALSE
+			else
+				if(!do_after(user, 10 SECONDS, buckledmeme))
+					return FALSE
+			buckledmeme.visible_message(span_notice("[user] unbuckles [buckledmeme] from [src]."),\
+				span_notice("[user] unbuckles you from [src]."),\
+				span_hear("You hear loose ropes."))
+		else
+			user.visible_message(span_notice("You unbuckle yourself from [src]."),\
+				span_notice("You unbuckle yourself from [src]."),\
+				span_hear("You hear loose ropes."))
+		add_fingerprint(user)
+		if(isliving(buckledmeme.pulledby))
+			var/mob/living/L = buckledmeme.pulledby
+			L.set_pull_offsets(buckledmeme, L.grab_state)
+	unbuckle_mob(buckled_mob)
+	return buckledmeme
+
+/obj/structure/chair/shibari_stand/user_buckle_mob(mob/living/buckledmeme, mob/user, check_loc = TRUE)
+	if(!is_user_buckle_possible(buckledmeme, user, check_loc))
+		return FALSE
+	add_fingerprint(user)
+
+	if(!ishuman(buckledmeme))
+		return FALSE
+
+	var/mob/living/carbon/human/hooman = buckledmeme
+	if(!(istype(hooman.w_uniform, /obj/item/clothing/under/shibari_fullbody)))
+		to_chat(user, span_warning("You'll need to completely tie their body!"))
+		return FALSE
+	if(!istype(user.get_active_held_item(), /obj/item/stack/shibari_rope))
+		to_chat(user, span_warning("You'll need to be holding shibari ropes to tie them to the stand!!"))
+		return FALSE
+
+	if(buckledmeme != user)
+		buckledmeme.visible_message(span_warning("[user] starts tying [buckledmeme] to [src]!"),\
+			span_userdanger("[user] starts tying you to [src]!"),\
+			span_hear("You hear ropes being tightened."))
+		if((HAS_TRAIT(user, TRAIT_RIGGER)))
+			if(!do_after(user, 5 SECONDS, buckledmeme))
+				return FALSE
+		else
+			if(!do_after(user, 10 SECONDS, buckledmeme))
+				return FALSE
+
+		if(!is_user_buckle_possible(buckledmeme, user, check_loc))
+			return FALSE
+
+		if(!(istype(hooman.w_uniform, /obj/item/clothing/under/shibari_fullbody)))
+			to_chat(user, span_warning("You'll need to completely tie their body!"))
+			return FALSE
+		if(!istype(user.get_active_held_item(), /obj/item/stack/shibari_rope))
+			to_chat(user, span_warning("You'll need to be holding shibari ropes to tie them to the stand!"))
+			return FALSE
+
+		if(buckle_mob(buckledmeme, check_loc = check_loc))
+			var/obj/item/stack/shibari_rope/rope = user.get_active_held_item()
+			ropee = new()
+			ropee.current_color = rope.current_color
+			ropee.update_icon_state()
+			ropee.update_icon()
+			rope.use(1)
+			add_overlay(shibari_shadow_overlay)
+			add_rope_overlays(ropee.current_color, hooman?.dna?.species?.mutant_bodyparts["taur"])
+			buckledmeme.visible_message(span_warning("[user] tied [buckledmeme] to [src]!"),\
+				span_userdanger("[user] tied you to [src]!"),\
+				span_hear("You hear ropes being completely tightened."))
```

Comment:
You'll need to use `\the` in there and many other places when you have `to [src]` otherwise it'll look really jank in-game.

Furthermore, these kind of messages should be going through a new proc that should get implemented that handles checking people's prefs to see if they should see those messages or not.

Also, in general, I don't see really anything that checks prefs anywhere in here.